Modules/System/Email/src/Account/EmailAccountImpl.Codeunit.al

```diff
@@ -101,8 +108,12 @@ codeunit 8889 "Email Account Impl."
             exit;
         end;
 
-        Commit();  // Commit the accounts deletion in order to prompt for new default account
-        if PromptNewDefaultAccountChoice(NewDefaultEmailAccount) then
+        NewDefaultEmailAccountSelected := false;
+        if not HideDialog then begin
+            Commit();  // Commit the accounts deletion in order to prompt for new default account
+            NewDefaultEmailAccountSelected := PromptNewDefaultAccountChoice(NewDefaultEmailAccount);
+        end;
+        if NewDefaultEmailAccountSelected then
             MakeDefault(NewDefaultEmailAccount)
         else
             EmailScenario.UnassignScenario(Enum::"Email Scenario"::Default); // remove the default scenario as it is pointing to a non-existent account
```

Comment:
If HideDialog is true, we would always remove the default scenario, even if the default account was not deleted, wouldn't we?

Reply:
@JesperSchulz no, we wouldn't:

Check out the line 102:
```al
        if AllEmailAccounts.Get(CurrentDefaultAccountId, Connector) then
            exit; // The default account was not deleted or it never existed
```
The procedure is exited if the default account still exists. (This is already tested for the UI tests.

Reply:
Perfect. All good then :-)